src/libYARP_OS/src/BottleImpl.cpp

```diff
@@ -804,8 +822,8 @@ bool StoreVocab::writeRaw(ConnectionWriter& writer)
 
 ConstString StoreDouble::toString() const
 {
-    char buf[512];
-    sprintf(buf, "%g", x);
+    char buf[YARP_DOUBLE_TO_STRING_MAX_LENGTH];    // -> see comment at the top of the file
+    snprintf(buf, YARP_DOUBLE_TO_STRING_MAX_LENGTH, "%.*g", DBL_DIG, x);
```

Comment:
This is pedantic, but `snprintf` is not guaranteed to set `/0` as the last character if the size of the produced string is greater than the size of the array, and static analysis tools will complain.
Please use:
```cpp
snprintf(buf, YARP_DOUBLE_TO_STRING_MAX_LENGTH - 1, "%.*g", DBL_DIG, x);
buf[YARP_DOUBLE_TO_STRING_MAX_LENGTH - 1] = '\0';

Reply:
@drdanz really it is not guaranteed? The documentations says that: `At most buf_size - 1 (for us is YARP_DOUBLE_TO_STRING_MAX_LENGTH - 1 ) characters are written. The resulting character string will be terminated with a null character.`
What am I misunderstanding here?

Reply:
`should` be...
Unfortunately have a look at https://stackoverflow.com/questions/7706936/is-snprintf-always-null-terminating

Reply:
Actually, at a second thought, I don't know if we are actually using `_snprintf` underneath
I think we should be using `std::snprintf` (that is guaranteed to have the terminating character) to ensure that we are not using some magic `#define`. I'm not 100% sure that it is supported in VS2013 though...

Reply:
I see. Anyway, shouldn't we always (or at least at the best of our possibilities) use C++ functions? I would use std::snprintf and also test it on different systems.